data_structures/graph/test_graph.py

```diff
@@ -0,0 +1,87 @@
+from graph import Graph, Node
+import pytest
+
+def test_add_node():
+    graph = Graph()
+    expected = 'apple' 
+    actual = graph.add_node('apple').value
+    assert actual == expected
+
+def test_size_empty():
+    graph = Graph()
+    expected = 0
+    actual = graph.size()
+    assert actual == expected
+
+
+def test_size():
+    graph = Graph()
+    graph.add_node('apple')
+    expected = 1
+    actual = graph.size()
+    assert actual == expected
+
+
+def test_add_edge_interloper_start():
+    graph = Graph()
+    start = Node('start')
+    end = graph.add_node('end')
+    with pytest.raises(KeyError):
+        graph.add_edge(start, end)
+
+
+
```

Comment:
All points off for extra white space!!!!! Im totally kidding, I just needed something to poke at because this submission is really good :D